### PR TITLE
Fixes #720 Issues related to current recording from devices

### LINKED ISSUE #720: Updates recording of currents provided by stimulating devices
This is a continuation of the discussion towards the end of #663.

The following needs to be updated:
- Setting the recordables: GetDefaults('..._generator')['recordables']
- Appropriate updates to unit test to include above
- For noise generator, set current = 0, when no target neuron connected
- For noise generator, keep the current value constant for the duration of noise_generator's dt
(solutions for above discussed in #663)

One outstanding issue is possibly regarding the current generated by AC source. Consider the following:
ac = nest.Create('ac_generator', 1, params = {'amplitude': 550.0, 'offset': 1000.0, 'frequency': 100.0, 'phase' : 0.0, 'origin' : 2.5, 'start' : 2.5, 'stop' : 40.0})

The values that are computed are:
```
t (ms) <-> I (pA)
4.9    <-> 0.0
5.0    <-> 1034.53478574
5.1    <-> 1068.93327846
5.2    <-> 1103.05972302
```    
min_delay = 1.0
```
t (ms) <-> V (mV)
5.9    <-> -70.0
6.0    <-> -70.0
6.1    <-> -69.58756241
6.2    <-> -69.16415163
```
Shouldn't the value at t = 5.0, be I = 1000.00.... i.e. offset + sin(t=0). 
ac_jump_v2
For Brian and NEURON, the current and voltage values are as follows:
(ignoring timestamps owing to differences in simulator execution)
```
Current:
0.0
1000.0
1034.53479
1068.93328
1103.05972

Voltage:
-70.0
-70.0
-69.90049751
-69.79805377
-69.69269695
-69.58446864
```
The absence of this value (1000.0) in NEST, I believe, causes a more abrupt change in both the current and voltage at the start of current injection as seen below, which is absent in the next cycle (starting at 15 ms). The change in Vm is more gradual in case of Brian and NEURON (as seen in the values above).

@heplesser , @gtrensch : Any thoughts on this? My current workaround for this is to compensate for the jump by adjusting the phase appropriately. But I believe there should be a more elegant fix, if this is infact considered an issue with the AC source.

### PR DESCRIPTION
**Pull request by appukuttan-shailesh**
Implements the following (as discussed in #720 and #663 ):

- Setting the recordables: GetDefaults('..._generator')['recordables']
- Appropriate updates to unit test to include above
- For noise generator, set current = 0, when no target neuron connected
- For noise generator, keep the current value constant for the duration of noise_generator's dt
- (solutions for above discussed in #663)

### CODE REVIEW

File: pynest/nest/tests/test_current_recording_generators.py [hunk 1]
```diff
@@ -85,6 +85,37 @@ def setUp(self):
                                          'stop': self.t_stop})
         nest.Connect(self.noise, self.neuron)
 
+    def test_GetRecordables(self):
+        """Check get recordables"""
+
+        try:
+            val = nest.GetDefaults('step_current_generator')['recordables'][0]
+            assert val == 'I', \
+                "Incorrect recordables ( %r ) for step current generator" % val
```
Comment (heplesser) [inline pynest/nest/tests/test_current_recording_generators.py]: Please use  
```
self.assertEqual(val, 'I', 'Incorrect recordables ({}) for step current generator'.format(val))
```
also for the tests below.

Comment (appukuttan-shailesh) [inline pynest/nest/tests/test_current_recording_generators.py]: Just to confirm, I would be replacing
``` python
try:
        val = nest.GetDefaults('step_current_generator')['recordables'][0]
        assert val == 'I', \
                "Incorrect recordables ( %r ) for step current generator" % val
except:
        raise
```
with
``` python
val = nest.GetDefaults('step_current_generator')['recordables'][0]
self.assertEqual(val, 'I', 'Incorrect recordables ({}) for step current generator'.format(val))
```
for each of the sub-tests

Comment (heplesser) [inline pynest/nest/tests/test_current_recording_generators.py]: Correct.

**Comment by heplesser** [comment]
@appukuttan-shailesh Could you pep8-ify the test file? Some lines are too long and there is trailing whitespace.

**Comment by appukuttan-shailesh** [comment]
Done.... looks good to go.

**Comment by heplesser** [comment]
This merge fixes current recording from noise generator and adds the recordables entry to other generators that support current recording.

**Pull request merged**